PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterCharacterScripts/Sound.rbxmx

```diff
@@ -0,0 +1,381 @@
+CreateNewSound("Swimming", 	"rbxasset://sounds/action_swim.mp3",						true,	1.6)
+CreateNewSound("Climbing", 	"rbxasset://sounds/action_footsteps_plastic.mp3",			true,	1)]]></ProtectedString>
+		</Properties>
+		<Item class="LocalScript" referent="RBX35b0a371f02e4ae2baca5961c8b67265">
+			<Properties>
+				<bool name="Disabled">false</bool>
+				<Content name="LinkedSource"><null></null></Content>
+				<string name="Name">LocalSound</string>
+				<string name="ScriptGuid"></string>
+				<ProtectedString name="Source"><![CDATA[--[[
+	Author: @spotco
+	This script runs locally for the player of the given humanoid.
+	This script triggers humanoid sound play/pause actions locally.
+	
+	The Playing/TimePosition properties of Sound objects bypass FilteringEnabled, so this triggers the sound
+	immediately for the player and is replicated to all other players.
+	
+	This script has been optimized to reduce network traffic (in comparison to the existing humanoid sound scripts.
+]]--
+
+--All sounds are referenced by this ID
+local SFX = {
+	Died = 0;
+	Running = 1;
+	Swimming = 2;
+	Climbing = 3,
+	Jumping = 4;
+	GettingUp = 5;
+	FreeFalling = 6;
+	FallingDown = 7;
+	Landing = 8;
+	Splash = 9;
+}
+
+local Humanoid = nil
+local Head = nil
+
+--SFX ID to Sound object
+local Sounds = {}
+
+do
+	local Figure = script.Parent.Parent
+	Head = Figure:WaitForChild("Head")
+	Humanoid = Figure:WaitForChild("Humanoid")
+	
+	Sounds[SFX.Died] = 			Head:WaitForChild("Died")
+	Sounds[SFX.Running] = 		Head:WaitForChild("Running")
+	Sounds[SFX.Swimming] = 		Head:WaitForChild("Swimming")
+	Sounds[SFX.Climbing] = 		Head:WaitForChild("Climbing")
+	Sounds[SFX.Jumping] = 		Head:WaitForChild("Jumping")
+	Sounds[SFX.GettingUp] = 	Head:WaitForChild("GettingUp")
+	Sounds[SFX.FreeFalling] = 	Head:WaitForChild("FreeFalling")
+	Sounds[SFX.Landing] = 		Head:WaitForChild("Landing")
+	Sounds[SFX.Splash] = 		Head:WaitForChild("Splash")
+end
+
+local Util
+Util = {
+	
+	--Define linear relationship between (pt1x,pt2x) and (pt2x,pt2y). Evaluate this like at x.
+	YForLineGivenXAndTwoPts = function(x,pt1x,pt1y,pt2x,pt2y)
+		--(y - y1)/(x - x1) = m
+		local m = (pt1y - pt2y) / (pt1x - pt2x)
+		--float b = pt1.y - m * pt1.x;
+		local b = (pt1y - m * pt1x)
+		return m * x + b
+	end;
+	
+	--Clamps the value of "val" between the "min" and "max"
+	Clamp = function(val,min,max)
+		return math.min(max,math.max(min,val))	
+	end;
+	
+	--Gets the horizontal (x,z) velocity magnitude of the given part
+	HorizontalVelocityMagnitude = function(Head)
+		local hVel = Head.Velocity + Vector3.new(0,-Head.Velocity.Y,0)
+		return hVel.magnitude	
+	end;
+	
+	--Gets the vertical (y) velocity magnitude of the given part
+	VerticalVelocityMagnitude = function(Head)
+		return math.abs(Head.Velocity.Y)
+	end;
+	
+	--Checks if Sound.Playing is usable
+	SoundIsV2 = function(sound)
+		return sound.IsLoaded
+	end;
+	
+	--Setting Playing/TimePosition values directly result in less network traffic than Play/Pause/Resume/Stop
```

Comment:
So it's 4 big savings:
1. Stopping only the active looped sounds (stopActiveLoopedSoundsExcept). Old script would stop all looped sounds (always stop swimming,running and climbing), even if they weren't playing.

2. Use of .Playing and .TimePosition if they're available (instead of Play/Pause/Resume/Stop). The Play methods are more network traffic because they've got additional replication things they need to do (update the internal "lastSetLuaTime", fire Played/Paused/Resumed/Stopped events)

3. Changes to the behaviour of the "Running" sound and "Falling" sound to not trigger quite as often. These are subtle behaviour changes, but what the player hears ingame should be indistinguishable.

4. Avoids any normally un-noticeable changes to Volume/Pitch (there were a couple of not-very-functional speed-to-pitch linear relationships in the old code, they're just set-once pitches now). This is only network traffic in non-filtering enabled games.